packages/protocol/contracts/common/Proxy.sol

```diff
@@ -1,10 +1,7 @@
-// TODO make Iproxy
```

Comment:
This import is somewhat important and not trivial to implement: This implementation of a proxy has a methods that should be external as public, this make us consume interfaces that are not fully consistent with the implementation. This will make the contract change bytecode, which is not something we can redeploy on mainnet for existing contracts.

Reply:
I assume this is a comment you left for future reference. I don't think I can merge this PR without resolving every conversation, so I'll copy/paste this as a comment below for future reference.